src/test/2.0/scala/com/holdenkarau/spark/testing/InputStreamTest.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.holdenkarau.spark.testing
+
+import com.holdenkarau.spark.testing.receiver.{InputStreamSuiteBase, InputStreamTestingContext}
+import org.apache.spark.SparkContext
+import org.apache.spark.streaming.dstream.DStream
+import org.apache.spark.streaming._
+import org.scalatest.FunSuite
+
+
+
+class InputStreamTest extends FunSuite with InputStreamSuiteBase {
+
+  test("verify success if inputs received within expected timeout") {
+    runTest(
+      inputDataForDstream = List(1, 3),
+      expectedResult = List(1, 3),  
+      generatorDelay = Seconds(0),
+      awaitResultsTimeout =  Seconds(10))
+  }
+
+  test("verify failure if inputs not received within expected timeout") {
+    val thrown = intercept[Exception] {
+      runTest(
+        inputDataForDstream = List(1, 3),
+        expectedResult = List(1, 3),
+        generatorDelay = Seconds(10),
+        awaitResultsTimeout =  Seconds(0))
+    }
+    assert(thrown.getMessage contains  "Futures timed out")
+  }
+
+  test("verify error raised if execpted results do not equal actual results obtained") {
+    val thrown = intercept[Exception] {
+      runTest(inputDataForDstream = List(1, 5), expectedResult = List(1, 3))
+    }
+    assert(thrown.getMessage contains  "!= expected result")
+  }
+
+  test("show that framework does not currently detect duplicates and order discrepancies when checking results") {
+    runTest(inputDataForDstream = List(2, 2, 1), expectedResult = List(1, 2))
+  }
+
+  def runTest(inputDataForDstream: List[Int],
```

Comment:
Hi .. sorry for delayed response (day job). 

re:  some tests that reflect anticipated 'real world' use cases:

The actual motivation for me to take a stab at enhancing your library was a self-education 
project in which 
I implemented an enhanced version of fileInputStream --  My version gives the file name and 
Seq[String] containing all lines of each file dropped into a folder being monitored.
The code under test is 
    here: 
        https://github.com/buildlackey/sparkStreamingFindDupWordsInFiles/blob/master/src/main/scala/CustomRecordReader.scala
        and 
    here: 
        https://github.com/buildlackey/sparkStreamingFindDupWordsInFiles/blob/master/src/main/scala/CustomInputFormat.scala

and the test that exercises the above code is here:
    https://github.com/buildlackey/sparkStreamingFindDupWordsInFiles/blob/master/src/test/scala/TextFileContentsWithFileNameSpec.scala

If you think a more 'real life' test case would be good, I could pull the above code in
to test out the feature I implemented.   Let me know and I will be happy to do that.